### PR TITLE
11798 add data attributes

### PR DESCRIPTION
**Pull request by bashmakovartur**
---

### Version updates:
```
ui-core/package.json
"version": "3.0.2"
"version": "3.1.0"

ui-shared/package.json
"version": "3.0.3"
"version": "3.1.0"
```
## :memo: packages/ui-core/CHANGELOG.md
# [3.1.0](https://github.com/MegafonWebLab/megafon-ui/compare/@megafon/ui-core@3.0.2...@megafon/ui-core@3.1.0) (2022-02-22)

### Bug Fixes

* **colors:** fix colors usage ([f2f59a8](https://github.com/MegafonWebLab/megafon-ui/commit/f2f59a81644c157394817af1531dabb1c0b67b19))
* **components:** better view of default props values ([1817ed5](https://github.com/MegafonWebLab/megafon-ui/commit/1817ed5cd17fd78a0911a89020138f05d7af9833))
* **contentarea:** add 'black' color prop value; deprecate 'default' color prop value; more examples ([e0b0c72](https://github.com/MegafonWebLab/megafon-ui/commit/e0b0c723a37f0e42ccb1c0d30df8ae36e9c82920))

### Features

* **banner:** add opacity for slides ([bb403a1](https://github.com/MegafonWebLab/megafon-ui/commit/bb403a1d38fd0c1efd3e1df3ed72ea4e5c772c40))
* **collapse:** changed collapse animation via setTimeout to requestAnimationFrame ([06d7b4f](https://github.com/MegafonWebLab/megafon-ui/commit/06d7b4ff0f3cdf3cc22fb1d45e8401e259636cf5))
* **ui-core:** add dataAttrs prop for interactive elements of components that did not have ([f495833](https://github.com/MegafonWebLab/megafon-ui/commit/f495833b33ac7823ae5992a3f7eb8b7cef82c036))

## :memo: packages/ui-shared/CHANGELOG.md
# [3.1.0](https://github.com/MegafonWebLab/megafon-ui/compare/@megafon/ui-shared@3.0.3...@megafon/ui-shared@3.1.0) (2022-02-22)

### Bug Fixes

* **videobanner:** update snapshots ([77de901](https://github.com/MegafonWebLab/megafon-ui/commit/77de901fa76793cf76dc776b3234b2d792befd9a))
* fix Breadcrumbs, ButtonLinkBox, Card, CardBox, Container, DownloadLink & VideoBanner snapshots ([e149e2d](https://github.com/MegafonWebLab/megafon-ui/commit/e149e2d40ececff0065ec1d800fb807fabfe8640))
* **breadcrumbs:** allow to pass items as array of JSX elements ([1ca1583](https://github.com/MegafonWebLab/megafon-ui/commit/1ca158390c7c58b5ff117ae8ad16268b043d5ddb))
* **colors:** fix colors usage ([f2f59a8](https://github.com/MegafonWebLab/megafon-ui/commit/f2f59a81644c157394817af1531dabb1c0b67b19))
* **components:** better view of default props values ([1817ed5](https://github.com/MegafonWebLab/megafon-ui/commit/1817ed5cd17fd78a0911a89020138f05d7af9833))

### Features

* **textwithiconitem:** add type string[] for text prop ([0df9680](https://github.com/MegafonWebLab/megafon-ui/commit/0df9680e10058ea24783fbc41cbce17e9b2a5594))
* **ui-shared:** add dataAttrs prop for interactive elements of components that did not have ([6b2fad8](https://github.com/MegafonWebLab/megafon-ui/commit/6b2fad84e72269d1fac0408e0b7e94202866b4aa))
* **videobanner:** added new classes for video ([d8b84a0](https://github.com/MegafonWebLab/megafon-ui/commit/d8b84a003103cf5f66102d354d2625e30a2c0d2a))

### CODE REVIEW

File: packages/ui-shared/src/components/Breadcrumbs/Breadcrumbs.tsx [hunk 1]
```diff
@@ -19,21 +19,29 @@ export const TextColor = {
...
+        {items.map(({ component, title, href }, i) => (
             <div className={cn('item', classes.item)} key={title}>
                 {component}
 
                 {!component && (
-                    <TextLink href={href} color={color}>
+                    <TextLink
+                        href={href}
+                        color={color}
+                        dataAttrs={{ root: { ...filterDataAttrs(dataAttrs?.link, i + 1) } }}
```
Comment (IGoRFonin) [inline packages/ui-shared/src/components/Breadcrumbs/Breadcrumbs.tsx:43]: любом случае передает, объект, может сделать проверку на наличие, посмотри в снапшоты как получается

Comment (IGoRFonin) [reply packages/ui-shared/src/components/Breadcrumbs/Breadcrumbs.tsx:43]: хотябы чтобы в root попадал `undefined`,  а не объект

File: packages/ui-shared/src/components/Instructions/Instructions.tsx [hunk 2]
```diff
@@ -115,29 +123,41 @@ const Instructions: React.FC<IInstructionsProps> = ({
...
-        (mediaUrl: string): JSX.Element => (
-            <video className={cn('swiper-img')} autoPlay muted loop playsInline>
+        (mediaUrl: string, index: number): JSX.Element => (
+            <video
+                loop
+                muted
+                autoPlay
+                playsInline
+                className={cn('swiper-img')}
+                {...filterDataAttrs(dataAttrs?.image, index)}
```
Comment (IGoRFonin) [inline packages/ui-shared/src/components/Instructions/Instructions.tsx:133]: index + 1, счетчики должны идти с единицы

File: packages/ui-shared/src/components/Instructions/Instructions.tsx [hunk 3]
```diff
@@ -115,29 +123,41 @@ const Instructions: React.FC<IInstructionsProps> = ({
...
+                    <SwiperSlide className={swiperSlideCn} key={i + mediaUrl}>
                         {isVideo ? (
-                            renderVideo(mediaUrl)
+                            renderVideo(mediaUrl, i)
                         ) : (
-                            <img className={cn('swiper-img', [instructionItemImg])} src={mediaUrl} alt="" />
+                            <img
+                                alt=""
+                                src={mediaUrl}
+                                {...filterDataAttrs(dataAttrs?.image, i)}
```
Comment (IGoRFonin) [inline packages/ui-shared/src/components/Instructions/Instructions.tsx:152]: i + 1

File: packages/ui-shared/src/components/Instructions/Instructions.tsx [hunk 4]
```diff
@@ -174,20 +194,21 @@ const Instructions: React.FC<IInstructionsProps> = ({
     const renderDesktopArticles = React.useCallback(
         (): JSX.Element => (
             <ul className={cn('articles-list', { 'text-after': !!text, desktop: true })}>
-                {instructionItems.map(({ title: itemTitle }, ind) => (
+                {instructionItems.map(({ title: itemTitle }, i) => (
                     // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
                     <li
-                        className={cn('articles-item', { active: slideIndex === ind }, [
-                            getActiveCustomClass(ind, slideIndex),
+                        {...filterDataAttrs(dataAttrs?.item, i)}
```
Comment (IGoRFonin) [inline packages/ui-shared/src/components/Instructions/Instructions.tsx:200]: i + 1

File: packages/ui-shared/src/components/Instructions/Instructions.tsx [hunk 5]
```diff
@@ -211,45 +233,53 @@ const Instructions: React.FC<IInstructionsProps> = ({
...
                     {instructionItems.map(
-                        ({ title: itemTitle }, ind) =>
-                            slideIndex === ind && (
-                                <div className={cn('articles-title', [mobileItemTitle])} data-index={ind} key={ind}>
+                        ({ title: itemTitle }, i) =>
+                            slideIndex === i && (
+                                <div
+                                    key={i}
+                                    data-index={i}
+                                    {...filterDataAttrs(dataAttrs?.mobileItemText)}
```
Comment (IGoRFonin) [inline packages/ui-shared/src/components/Instructions/Instructions.tsx:241]: добавить индекс

File: packages/ui-shared/src/components/Instructions/Instructions.tsx [hunk 6]
```diff
@@ -211,45 +233,53 @@ const Instructions: React.FC<IInstructionsProps> = ({
...
                     )}
                 </div>
                 <ul className={cn('articles-dots', { 'text-after': !!text })}>
-                    {instructionItems.map((_item, ind) => (
+                    {instructionItems.map((_item, i) => (
                         <div
-                            key={ind}
-                            className={cn('articles-dot', { active: slideIndex === ind }, [
-                                getActiveCustomClass(ind, slideIndex),
+                            {...filterDataAttrs(dataAttrs?.item, i)}
```
Comment (IGoRFonin) [inline packages/ui-shared/src/components/Instructions/Instructions.tsx:252]: i + 1

File: packages/ui-shared/src/components/Partners/Partners.tsx [hunk 7]
```diff
@@ -108,16 +117,16 @@ const Partners: React.FC<IPartnersProps> = ({
             >
                 {topRow.map((item, i) => (
                     <div key={i + item.src} className={cn('slide')}>
-                        {renderItem(item)}
-                        {renderItem(bottomRow[i])}
+                        {renderItem(i, item)}
+                        {renderItem(i, bottomRow[i])}
```
Comment (IGoRFonin) [inline packages/ui-shared/src/components/Partners/Partners.tsx:121]: лучше по общепринятому индекс передавать вторым аргументом

File: packages/ui-shared/src/components/VideoBlock/VideoBlock.tsx [hunk 8]
```diff
@@ -97,6 +100,7 @@ const VideoBlock: React.FC<IVideoBlockProps> = ({
                 <div className={cn('description', [classes.description])}>{description}</div>
                 {buttonTitle && (
                     <Button
+                        dataAttrs={{ root: dataAttrs?.button }}
```
Comment (IGoRFonin) [inline packages/ui-shared/src/components/VideoBlock/VideoBlock.tsx:103]: элемент списка, нужен индекс

Comment (bashmakovartur) [reply packages/ui-shared/src/components/VideoBlock/VideoBlock.tsx:103]: Не, это не список

**Pull request merged**